crates/fuel-core/src/query/balance/asset_query.rs

```diff
@@ -95,10 +99,9 @@ impl<'a> AssetsQuery<'a> {
     /// # Note: The resources of different type are not grouped by the `asset_id`.
     // TODO: Optimize this by creating an index
     //  https://github.com/FuelLabs/fuel-core/issues/588
-    pub fn unspent_resources(
-        &self,
-    ) -> impl Iterator<Item = StorageResult<Resource>> + '_ {
-        let coins_iter = CoinQueryContext(self.database)
+    pub fn unspent_resources(&self) -> BoxedIter<StorageResult<Resource>> {
```

Comment:
The output type should be `impl Iterator<Item = StorageResult<Resource>>`.

```
             .into_boxed()
             .collect::<Vec<_>>()
             .into_iter()
             .into_boxed()
```

Should definitely be re-worked=) I guess the previous implementation should work for you

Reply:
The output type of `impl Iterator<Item = StorageResult<Resource>>` can't be used as a return type in a trait definition, and referencing the iterator type directly is nearly impossible because it ends up being long enough that the compiler will dump its type name into a `.txt` file when compiling, is there another way around this?

Reply:
It is not a trait, it is a public method=)

Reply:
Oh I had this confused from some of the other `impl Iterator`'s which I changed, thanks for fixing this!